cli/azd/cmd/sets.go

```diff
@@ -26,36 +26,33 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func newWriter(cmd *cobra.Command) io.Writer {
-	writer := cmd.OutOrStdout()
-
-	if os.Getenv("NO_COLOR") != "" {
-		writer = colorable.NewNonColorable(writer)
-	}
+func newWriterFromConsole(console input.Console) io.Writer {
+	return console.GetWriter()
+}
 
-	return writer
+func newFormatterFromConsole(console input.Console) output.Formatter {
+	return console.GetFormatter()
 }
 
 func newConsoleFromOptions(
 	rootOptions *internal.GlobalCommandOptions,
 	formatter output.Formatter,
-	writer io.Writer,
 	cmd *cobra.Command,
 ) input.Console {
-	// NOTE: There is a similar version of this code in pkg/commands/builder.go that exists while we transition
-	// from the old plan of passing everything via a context to the new plan of wiring everything up explicitly.
-	//
-	// If you make changes to this logic here, also take a look over there to make the same changes.
-
-	isTerminal := cmd.OutOrStdout() == os.Stdout &&
-		cmd.InOrStdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()) &&
-		isatty.IsTerminal(os.Stdout.Fd())
-
+	writer := cmd.OutOrStdout()
 	// When using JSON formatting, we want to ensure we always write messages from the console to stderr.
 	if formatter != nil && formatter.Kind() == output.JsonFormat {
 		writer = cmd.ErrOrStderr()
 	}
 
+	if os.Getenv("NO_COLOR") != "" {
+		writer = colorable.NewNonColorable(writer)
+	}
```

Comment:
This is intentionally changed to be after checking for formatted output, so that messages printed out to json would still be stripped of colored output. In reality, it probably doesn't matter.